### PR TITLE
Cleanup Excelx

### PR DESCRIPTION
**Pull request by stevendaniels**

### CODE REVIEW

File: lib/roo/excelx.rb [hunk 1]
```diff
@@ -1,667 +1,491 @@
...
-      if type = EXCEPTIONAL_FORMATS[format]
-        type
-      elsif format.include?('#')
-        :float
-      elsif !format.match(/d+(?![\]])/).nil? || format.include?('y')
-        if format.include?('h') || format.include?('s')
-          :datetime
+module Roo
+  class Excelx < Roo::Base
+    autoload :Workbook, 'roo/excelx/workbook'
```
Comment (simonoff) [inline lib/roo/excelx.rb]: I think we can require this modules except of autoload. No need to make autoload there.

File: lib/roo/excelx.rb [hunk 2]
```diff
@@ -1,667 +1,491 @@
...
+    autoload :Comments, 'roo/excelx/comments'
+    autoload :SheetDoc, 'roo/excelx/sheet_doc'
+
+    module Format
+      EXCEPTIONAL_FORMATS = {
+        'h:mm am/pm' => :date,
+        'h:mm:ss am/pm' => :date
+      }
+
+      STANDARD_FORMATS = {
```
Comment (simonoff) [inline lib/roo/excelx.rb]: I think you can freeze all this strings.

File: lib/roo/excelx/cell.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        @excelx_type = excelx_type
+        @excelx_value = excelx_value
+        @style = style
+        @value = type_cast_value(value)
+        @value = Roo::Link.new(hyperlink, @value.to_s) if hyperlink
+        @coordinate = coordinate
+      end
+
+      def type
+        if @formula
```
Comment (simonoff) [inline lib/roo/excelx/cell.rb]: I think will be better to use `case` instead of `if`

File: lib/roo/excelx/cell.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        case @type
+        when :float, :percentage
+          value.to_f
+        when :date
+          create_date(@base_date + value.to_i)
+        when :datetime
+          create_datetime(@base_date + value.to_f.round(6))
+        when :time
+          value.to_f * 86_400
+        when :string
```
Comment (simonoff) [inline lib/roo/excelx/cell.rb]: We can skip this variant because in any case it returns `value`

File: lib/roo/excelx/comments.rb [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
-    def extract_comments
-      if doc_exists?
-        Hash[doc.xpath("//comments/commentList/comment").map do |comment|
-          value = (comment.at_xpath('./text/r/t') || comment.at_xpath('./text/t')).text
-          [::Roo::Utils.ref_to_key(comment.attributes['ref'].to_s), value]
-        end]
-      else
-        {}
+      def extract_comments
+        if doc_exists?
```
Comment (simonoff) [inline lib/roo/excelx/comments.rb]: I think we can avoid this `if`

```ruby
return {} unless doc_exists?
Hash[doc.xpath('//comments/commentList/comment').map do |comment|
  value = (comment.at_xpath('./text/r/t') || comment.at_xpath('./text/t')).text
  [::Roo::Utils.ref_to_key(comment.attributes['ref'].to_s), value]
end]
```

**Comment by stevendaniels** [comment]
@simonoff Thanks. I'll work on those changes.

**Pull request merged**